### PR TITLE
Adding docs and specs to AdminMenu.nested_menu_items

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Documentation Update

## Description

This relates to PR forem/forem#16847 which addresses issue
forem/forem#16842.

The goal of this PR is to help me develop an understanding of the
AdminMenu so I can further extend it and better understand it.

There's also a bit of knowledge sharing that I'm looking for, and the
comments and tests are there to help "confirm" that knowledge.  There
might be more that I'd consider regarding a refactor, but I want to make
sure I'm on the right path before I go further.

tl;dr - this is the smallest commit I can make to begin to tease out an
understanding of the method I put under test.

## Related Tickets & Documents

- Related Issue forem/forem#16842

## QA Instructions, Screenshots, Recordings

Fire up the site and poke around in the admin navigation?  This should be a non-change.1

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

File: app/models/admin_menu.rb [hunk 1]
```diff
@@ -77,8 +87,14 @@ def self.nested_menu_items(scope_name, nav_item)
 
       items[:children].each do |child|
         return items if child[:controller] == nav_item
+        # NOTE: [@jeremyf] trying to puzzle this one out.  My read is that if the "grandchild" of the
+        # scope matches the controller, return the parent node (e.g. the item).
       end
     end
+
+    # Because we're using each loops, with short-circuiting returns, we need to make sure we don't
+    # retrun the results of `items[:children].each`, which will be `items[:children]`.
```
Comment (Ridhwana) [inline app/models/admin_menu.rb:96]:
```suggestion
    # Because we're using each loops, with short-circuiting returns, we need to make sure we don't
    # return the results of `items[:children].each`, which will be `items[:children]`.
```

File: app/models/admin_menu.rb [hunk 2]
```diff
@@ -77,8 +87,14 @@ def self.nested_menu_items(scope_name, nav_item)
 
       items[:children].each do |child|
         return items if child[:controller] == nav_item
+        # NOTE: [@jeremyf] trying to puzzle this one out.  My read is that if the "grandchild" of the
```
Comment (Ridhwana) [inline app/models/admin_menu.rb:90]: I suspect a regression occurred in the last couple of months. Here's some more detail: https://www.loom.com/share/7e0ce358120e4f3fb6df59f2cfad6183

It seems to not be a regression on your PR, so either you or I can look into this in a separate PR.

File: app/models/admin_menu.rb [hunk 3]
```diff
@@ -67,6 +67,16 @@ def self.navigation_items
     feature_flagged_menu_items
   end
 
+  # Return the Menu item that corresponds to the nav_item within the given named scope.
+  #
+  # @param scope_name [String]
+  # @param nav_item [String]
```
[the remark this answers is not in the document]
Comment (jeremyf) [reply app/models/admin_menu.rb:73]: Sure.  I'll ping you when I have that.

File: app/models/admin_menu.rb (hunk 2 above)
Comment (jeremyf) [reply app/models/admin_menu.rb:90]: That Loom is super helpful.  I'm prepared to dive deeper into the AdminMenu to resolve this regression.  I have some ideas and with this context, I am well equipped to dig deeper.

I'm going to write up an issue (and link to your Loom).  This way we can merge this PR and also not lose track of fixing that regression.

Comment (Ridhwana) [reply app/models/admin_menu.rb:90]: Sounds good, thanks Jeremy!

Comment (jeremyf) [reply app/models/admin_menu.rb:94]: I added https://github.com/forem/forem/issues/16901 to capture this regression.

File: app/models/admin_menu.rb [hunk 4]
```diff
@@ -76,11 +91,28 @@ def self.nested_menu_items(scope_name, nav_item)
...
 
+  # @param request [#path] the request object (which must respond to `#path`)
+  # @return [NilClass] when we don't have a menu representation of the nav_item
+  # @return [Hash] the representation of the nav_item
+  #
+  # @see {.nested_menu_items} for implementation details
+  #
+  # @note This method assumes that the last two slugs of the request's path are the relevant
+  #       information for determining which menu item to return.  In other words, strongly consider
+  #       the impact of having admin routes who's paths are comprised of more than 3 slugs (.e.g. we
```
Comment (juliannatetreault) [inline app/models/admin_menu.rb:113]: Quick, non-blocking suggestion:
```suggestion
  #       the impact of having admin routes whose paths are comprised of more than 3 slugs (.e.g. we
```

**Review by Ridhwana — APPROVED** [review]
